src/GuiBase/KeyFrameEditor/KeyFramedValueEditorWidgetRuler.h

```diff
@@ -0,0 +1,70 @@
+#ifndef RADIUMENGINE_QEDITORWIDGETRULER_H_
+#define RADIUMENGINE_QEDITORWIDGETRULER_H_
+
+#include <GuiBase/Timeline/Configurations.h>
+
+#include <QWidget>
+
+#include <Core/CoreMacros.hpp>
+
+namespace Ra::GuiBase {
+
+class QEditorWidgetRuler : public QWidget
+{
+    Q_OBJECT
+  public:
+    explicit QEditorWidgetRuler( QWidget* parent = nullptr );
+
+    int drawRuler( int newWidth, int height );
+
+    int getNbIntervalTime();
+    Scalar getStepTime();
+    Scalar getPixPerSec();
+    Scalar getZeroTime();
+    Scalar getMaxTime();
+    void setMaxTime( Scalar maxTime );
+
+    int getNbIntervalValue();
+    Scalar getStepValue();
+    Scalar getPixPerValue();
+    Scalar getZeroValue();
+    Scalar getMaxValue();
+
+  signals:
+    void stepChanged( Scalar step );
+
+  public slots:
+    void onDrawRuler( int width, int height );
+
+  private:
+    static const int s_nbSteps      = 13;
+    const Scalar m_steps[s_nbSteps] = {0.01_ra,
```

Comment:
Is it an instance attribute or a class attribute ? if instances can't change the value of this, make it a static const ... attribute.